src/octoprint/plugins/softwareupdate/static/js/softwareupdate.js

```diff
@@ -99,6 +108,10 @@ $(function () {
             );
         });
 
+        // Hack to remove automatically added Cancel button
+        // See https://github.com/sciactive/pnotify/issues/141
+        PNotify.prototype.options.confirm.buttons = [];
+
```

Comment:
This just tickled my memory, and then I remember that I had already added a workaround for that:

https://github.com/OctoPrint/OctoPrint/blob/1d4043736774dd9b4ea3eb8381100a38d8943fd4/src/octoprint/static/js/app/main.js#L283-L319

So just use `PNotify.singleButtonNotify` if you want to get rid of this button. What you do here will influence the defaults for any notifications spawned after first invocation of this line and thus might have nasty side effects.

Reply:
So I tried this and pushed the changes up, but it has the nasty side effect of not handling the Pnotify.update at line 1361. The initial popup doesn't have the default cancel button, but as soon as the update happens the button comes back. Maybe a better hack would be to add a hidden class to the empty cancel button?

Reply:
I switched it over to a hidden button instead. Achieves the same goal with all the hackery.

Reply:
Good solution, I didn't realize this would break on updating the popup. Might need a solution in core but for now this works.